libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/TilesetDescriptorFactory.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.navigation.core.navigator
+
+import com.mapbox.common.TilesetDescriptor
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigator.CacheHandle
+
+private typealias NativeTilesetDescriptorFactory = com.mapbox.navigator.TilesetDescriptorFactory
+
+/**
+ * A factory to build navigation [TilesetDescriptor]
+ */
+class TilesetDescriptorFactory internal constructor(
```

Comment:
Is there an option to expose just:
```kotlin
fun build(): TilesetDescriptor
```

that will take the current options from the `MapboxNavigation`/`Navigator`?

Reply:
no, we need 3 functions:
- build descriptor with a passed version+dataset
- build with server version
- build with local version used by navigator

Reply:
Why can't we fulfill that local/server dataset name based on the current routing tiles options?

Reply:
what if customer need another dataset for offline regions?

Reply:
we can make a `dataset` param nullable and pass null by default. 
if null -> use from options
non null -> use it